internal/cbor/types_test.go

```diff
@@ -128,12 +332,308 @@ func TestAppendInt(t *testing.T) {
+	if !bytes.Equal(got, want) {
+		t.Errorf("AppendUints8(%v)\ngot:  0x%s\nwant: 0x%s",
+			array, hex.EncodeToString(got),
+			hex.EncodeToString(want))
+	}
+}
+
+func TestAppendUint32(t *testing.T) {
+	for _, tc := range unsignedIntegerTestCases {
+		if tc.val > math.MaxUint32 {
+			continue
+		}
+		s := enc.AppendUint32([]byte{}, uint32(tc.val))
+		got := string(s)
+		want := tc.bigbinary
+		if got != want {
+			t.Errorf("AppendUint32(0x%x)=0x%s, want: 0x%s",
+				tc.val, hex.EncodeToString(s),
+				hex.EncodeToString([]byte(want)))
+		}
+	}
+}
+func TestAppendUints32(t *testing.T) {
+	array := make([]uint32, 0)
+	want := make([]byte, 0)
+	want = append(want, 0x98) // start array
+	want = append(want, 0x1f) // for unsigned  32-bit elements
+	for _, tc := range unsignedIntegerTestCases {
+		if tc.val > math.MaxUint32 {
+			continue
+		}
+		array = append(array, uint32(tc.val))
+		want = append(want, tc.bigbinary...)
+	}
+
+	got := enc.AppendUints32([]byte{}, array)
+	if !bytes.Equal(got, want) {
+		t.Errorf("AppendUints32(%v)\ngot:  0x%s\nwant: 0x%s",
+			array, hex.EncodeToString(got),
+			hex.EncodeToString(want))
+	}
+
+	// now empty array case
+	array = make([]uint32, 0)
+	want = make([]byte, 0)
+	want = append(want, 0x9f) // start and end array
+	want = append(want, 0xff) // for empty array
+	got = enc.AppendUints32([]byte{}, array)
+	if !bytes.Equal(got, want) {
+		t.Errorf("AppendUints32(%v)\ngot:  0x%s\nwant: 0x%s",
+			array, hex.EncodeToString(got),
+			hex.EncodeToString(want))
+	}
+}
+
+func TestAppendUint64(t *testing.T) {
+	for _, tc := range unsignedIntegerTestCases {
+		s := enc.AppendUint64([]byte{}, uint64(tc.val))
+		got := string(s)
+		want := tc.bigbinary
+		if got != want {
+			t.Errorf("AppendUint64(0x%x)=0x%s, want: 0x%s",
+				tc.val, hex.EncodeToString(s),
+				hex.EncodeToString([]byte(want)))
+		}
+	}
+}
+func TestAppendUints64(t *testing.T) {
+	array := make([]uint64, 0)
+	want := make([]byte, 0)
+	want = append(want, 0x98) // start array
+	want = append(want, 0x21) // for unsigned 64-bit elements
+	for _, tc := range unsignedIntegerTestCases {
+		array = append(array, uint64(tc.val))
+		want = append(want, tc.bigbinary...)
+	}
+
+	got := enc.AppendUints64([]byte{}, array)
+	if !bytes.Equal(got, want) {
+		t.Errorf("AppendUints64(%v)\ngot:  0x%s\nwant: 0x%s",
+			array, hex.EncodeToString(got),
+			hex.EncodeToString(want))
+	}
+
+	// now empty array case
+	array = make([]uint64, 0)
+	want = make([]byte, 0)
+	want = append(want, 0x9f) // start and end array
+	want = append(want, 0xff) // for empty array
+	got = enc.AppendUints64([]byte{}, array)
+	if !bytes.Equal(got, want) {
+		t.Errorf("AppendUints64(%v)\ngot:  0x%s\nwant: 0x%s",
+			array, hex.EncodeToString(got),
+			hex.EncodeToString(want))
+	}
+}
+
+func TestAppendUint(t *testing.T) {
+	for _, tc := range unsignedIntegerTestCases {
+		s := enc.AppendUint([]byte{}, tc.val)
+		got := string(s)
+		want := tc.bigbinary
+		if tc.val == math.MaxUint64 {
```

Comment:
The cool optimization that assigned a single-byte representation to Max-Unsigned-Native-Int